package.json

```diff
@@ -5,18 +5,43 @@
   "type": "module",
   "exports": {
     ".": {
-      "types": "./dist/index.d.ts",
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/node/index.js"
+      },
+      "require": {
+        "types": "./dist/index.d.cts",
+        "default": "./dist/node/index.cjs"
+      },
       "node": {
-        "import": "./dist/node/index.js",
-        "require": "./dist/node/index.cjs"
+        "import": {
+          "types": "./dist/index.d.ts",
+          "default": "./dist/node/index.js"
+        },
+        "require": {
+          "types": "./dist/index.d.cts",
+          "default": "./dist/node/index.cjs"
+        }
       },
       "browser": {
-        "import": "./dist/browser/index.js",
-        "require": "./dist/browser/index.cjs"
+        "import": {
+          "types": "./dist/index.d.ts",
+          "default": "./dist/browser/index.js"
+        },
+        "require": {
+          "types": "./dist/index.d.cts",
+          "default": "./dist/browser/index.cjs"
+        }
       },
       "worker": {
-        "import": "./dist/browser/index.js",
-        "require": "./dist/browser/index.cjs"
+        "import": {
+          "types": "./dist/index.d.ts",
+          "default": "./dist/browser/index.js"
+        },
+        "require": {
+          "types": "./dist/index.d.cts",
+          "default": "./dist/browser/index.cjs"
+        }
```

Comment:
Why not just add `types` ad the base level, above `import` and `require`? The types file can potentially be the same for both, as in my other comment.